metrics.py

```diff
@@ -76,6 +72,29 @@ def calculate_for_sample(self, networks_sample: np.ndarray | Collection[nx.Graph
             result[:, i] = self.calculate(network)
         return result
 
+    def calculate_mple_regressors(self, observed_network: np.ndarray | nx.Graph, edges_indices_lims: tuple[int]):
```

Comment:
Note for future, and something to have in mind - 

If we want users to be able to easily add custom metrics, then we have to understand what is the bare minimum they're expected to implement.

Right now it seems that a metric has to support - 
`calc_change_score`
`calculate_for_sample`
`calculate_mple_regressors`

but I'm guessing there's some overlap between these functions, and if my graphs aren't super big + I don't care about distributed runs, maybe all I need is to implement something more basic, such as `calc_change_score` + `calculate_for_sample` and that's it.

Reply:
now that I think of it, a user doesn't really need to implement `calculate_mple_regressors` because it internally only needs `calc_change_score` , so custom metrics can simply rely on the parent Metric class

Reply:
Technically, if you don't care about performance at all, all you need is `calculate` (for a single network). Everything else can build on that (and I think that this is currently the case, if you inherit from `Metric` and implement `calculate`, which is indeed an abstract method so you sort of must do that, you'll be fine)